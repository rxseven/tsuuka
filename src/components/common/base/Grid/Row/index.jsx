import cx from 'classnames';
import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  alignment: PropTypes.string,
  children: PropTypes.node.isRequired
});

const defaultProps = {
  alignment: ''
};

function Row({ alignment, children }) {
  return <div className={cx('row', !!alignment && alignment)}>{children}</div>;
}

Row.propTypes = propTypes;
Row.defaultProps = defaultProps;

export default Row;
