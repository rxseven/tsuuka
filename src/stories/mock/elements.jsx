import React from 'react';

// Placeholder content
const content = <div>Placeholder content</div>;

export default { content };
