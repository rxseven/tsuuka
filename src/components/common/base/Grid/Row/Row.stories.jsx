import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Row from './index';

const { content } = mock.elements;
const title = 'Components/Common/Base/Grid/Row';
const props = { children: content };

storiesOf(title, module)
  .add('default', () => <Row {...props} />, {
    notes: notes.minimal
  })
  .add('alignment', () => <Row {...props} alignment="justify-content-end" />);
