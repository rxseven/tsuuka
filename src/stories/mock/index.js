import elements from './elements';
import text from './text';

export default {
  elements,
  text
};
