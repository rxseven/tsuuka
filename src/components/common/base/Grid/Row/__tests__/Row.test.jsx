import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import mock from 'tests/mock';
import Row from '../index';

describe('<Row />', () => {
  // Arrange
  const props = { children: mock.elements.children };
  const component = <Row {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
