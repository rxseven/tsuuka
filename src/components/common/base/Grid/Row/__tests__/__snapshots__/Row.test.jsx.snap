// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> Snapshot tests should render correctly 1`] = `
<div
  className="row"
>
  <div>
    Content
  </div>
</div>
`;
